### PR TITLE
Use buffers around read and write sockets calls

### PR DESCRIPTION
**Pull request by prasad-joshi**

### CODE REVIEW

File: usr/net_is.c [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+}
+
+int net_is_last_error(struct net_is* isp)
+{
+	return isp->error;
+}
+
+bool net_is_has_data(struct net_is* isp)
+{
+#if 1
```
Comment (dmehta85) [inline usr/net_is.c]: Please remove.

File: usr/iscsi/iscsi_tcp.c [hunk 2]
```diff
@@ -494,26 +533,46 @@ static int iscsi_tcp_conn_login_complete(struct iscsi_connection *conn)
...
+	if (net_os_closed(conn->out_stream)) {
+		errno = ECONNRESET;
+		return -1;
+	}
+	errno = net_os_last_error(conn->out_stream);
+	return -1;
+#endif
 }
 
 static void iscsi_tcp_write_end(struct iscsi_connection *conn)
```
Comment (dmehta85) [inline usr/iscsi/iscsi_tcp.c]: Is this function never gets called?

File: usr/iscsi/iscsi_tcp.c [hunk 3]
```diff
@@ -265,6 +268,12 @@ static void accept_connection(int afd, int events, void *data)
 	tcp_conn->fd = fd;
 	conn->tp = &iscsi_tcp;
 
+	conn->in_stream = net_is_alloc(fd, (1ul << 20) * 4);
+	assert(conn->in_stream);
+
+	conn->out_stream = net_os_alloc(fd, (1ul << 10) * 128);
```
[the remark this answers is not in the document]
Comment (prasad-joshi) [reply usr/iscsi/iscsi_tcp.c]: done

File: usr/net_is.c (hunk 1 above)
Comment (prasad-joshi) [reply usr/net_is.c]: done

File: usr/iscsi/iscsi_tcp.c (hunk 2 above)
Comment (prasad-joshi) [reply usr/iscsi/iscsi_tcp.c]: we cannot remove this function. Look for calls to ep_write_end().

File: usr/iscsi/iscsi_tcp.c [hunk 4]
```diff
@@ -48,6 +49,8 @@ static long nop_ttt;
 static int listen_fds[8];
 static struct iscsi_transport iscsi_tcp;
 
+#define USE_NET_IN_STREAM 1
```
[the remark this answers is not in the document]
Comment (prasad-joshi) [reply usr/iscsi/iscsi_tcp.c]: I am going to keep these macros for now. There is one more change required fairness in TGT. We will remove this macros in that commit.

**Pull request merged**